### PR TITLE
Add note to README on non-expiring access token fix

### LINKED ISSUE #25: expires_in=0 ends up with a 1 year token (will this burn is in 1y)
> From my (albeit very quick) testing in 7.49.3, setting the `expires_in` field to `0` in the `POST /access/api/v1/tokens` request works to create a non-expiring token. There's no `expires_in` field when `GET /access/api/v1/tokens` and no expiry date in the web UI. I am able to revoke by `DELETE /access/api/v1/tokens/{id}`.
>
> When I create a token with `expires_in` set to a non-zero value (say 900 = 15 min), I was not able to revoke it which is correct behavior.
> 
> ```json
> {
>     "code": "BAD_REQUEST",
>     "message": "Token not revocable",
>     "detail": "Token not revocable. Token expirationTimeMillis: 1673028200098, issuedAtMillis: 1673027300098, revocableExpiryThresholdMillis: 21600000"
> }
> ```

_Originally posted by @alexhung in https://github.com/jfrog/artifactory-secrets-plugin/issues/22#issuecomment-1373974190_
     

We may have to add something to the readme if we can figure out what version that "non-expiring" token thing was fixed in. In the one that the user is referencing (7.33.12), even when expires_in=0, the token still ends up with a one year expiration, but it *does* allow the token to be revoked (curl -X DELETE). I am afraid that our current setup will silently fail 1year from setup due to the "default" 1 year token expiration being applied.

### PR DESCRIPTION
**Pull request by alexhung**
Close #25

### CODE REVIEW

**Review by Turhsus — DISMISSED** [review]
LGTM, but left a nitpick

File: README.md [hunk 1]
```diff
@@ -15,8 +15,14 @@ Using this plugin, you can limit the accidental exposure window of Artifactory t
...
 
-This backend creates access tokens in Artifactory using the admin credentials provided. Note that if you
-provide non-admin credentials, then the "username" must match the username of the credential owner.
+This backend creates access tokens in Artifactory using the admin credentials provided. Note that if you provide non-admin credentials, then the "username" must match the username of the credential owner.
+
+> **Note**
+> Prior to Artifactory 7.42.1, admin access token was created with the system token expiration (default to 1 hour) even when `expires_in` API field is set to 0. In 7.42.1, admin token expiration no longer constrained by system configuration and therefore can be set to non-expiring. See section ["Generate a Non-expiry Admin Token without Changing the Configuration"](https://www.jfrog.com/confluence/display/JFROG/Artifactory+Release+Notes#ArtifactoryReleaseNotes-Artifactory7.42.1Cloud) in the release note.
+>
+> Therefore if you created access token(s) with Artifactory prior to 7.42.1, the tokens will have a 1 year expiration time (or whatever value set in Artifactory configuration) and will become unusable silently when it expires.
```
Comment (Turhsus) [inline README.md:23]: nit: `or whatever value _is_ set in _the_ artifactory configuration`

File: README.md [hunk 2]
```diff
@@ -15,8 +15,14 @@ Using this plugin, you can limit the accidental exposure window of Artifactory t
...
+This backend creates access tokens in Artifactory using the admin credentials provided. Note that if you provide non-admin credentials, then the "username" must match the username of the credential owner.
+
+> **Note**
+> Prior to Artifactory 7.42.1, admin access token was created with the system token expiration (default to 1 hour) even when `expires_in` API field is set to 0. In 7.42.1, admin token expiration no longer constrained by system configuration and therefore can be set to non-expiring. See section ["Generate a Non-expiry Admin Token without Changing the Configuration"](https://www.jfrog.com/confluence/display/JFROG/Artifactory+Release+Notes#ArtifactoryReleaseNotes-Artifactory7.42.1Cloud) in the release note.
+>
+> Therefore if you created access token(s) with Artifactory prior to 7.42.1, the tokens will have a 1 year expiration time (or whatever value is set in the Artifactory configuration) and will become unusable silently when it expires.
+>
+> We suggest upgrade your Artifactory to 7.42.1 or later (if possible) and rotate your tokens to get new, non-expiring tokens. Or set reminders to ensure you rotate your tokens before expiration.
```
Comment (shrajfr12) [inline README.md:25]: changing "upgrade"  to "upgrading" looks better

**Review by shrajfr12 — APPROVED** [review]

**Review by Turhsus — APPROVED** [review]
LGTM

**Review by danielmkn — APPROVED** [review]

**Review by mritunjaykumar — APPROVED** [review]
LGTM

**Pull request merged**